autotest/autotest_services/base.py

```diff
@@ -1138,3 +1138,93 @@ def setUp(self):
+        if part == "xml":
+            return "xml"
+        elif part == "raster":
+            return "tif"    
+
+    @staticmethod
+    def parseFileName(src) : 
+        try : 
+            with file( src ) as fid : 
+                return fid.read() 
+        except Exception as e : 
+            raise XMLParseError , "Failed to parse the \"%s\" file! %s" % ( src , str(e) ) 
+
+    def parse( self, src) :
+        return  self.parseFileName(src) 
+
+
+    def testXMLComparison(self):
+        
+        expected_path= os.path.join(
+            self.getExpectedFileDir(), self.getExpectedFileName('xml')
+        )
+
+        expectedString= self.parse(expected_path)
+        doc = etree.fromstring(expectedString)
+        encodedText= ' '.join(e.text for e in doc.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'}))
+       
+        # replace the encoded data so it compare other nodes in the xml files
+        tree = etree.fromstring(self.prepareXMLData(self.getXMLData()))
+        for node in tree.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'}):
+            node.text = encodedText
+            
+            
+        self.response.content = etree.tostring(tree, encoding="ISO-8859-1")
+
+        super(WPS10XMLComparison, self).testXMLComparison()
+        
+
+    def testBinaryComparisonRaster(self):
+
+        response_path = os.path.join(
+            self.getResponseFileDir(), self.getResponseFileName("raster")
+        )
+        expected_path= os.path.join(
+            self.getExpectedFileDir(), self.getExpectedFileName("raster")
+        )
+        # creates a response image that contains the encoded text of the response xml file 
+        doc = etree.fromstring( self.prepareXMLData(self.getXMLData()))
+        encodedText= ' '.join(e.text for e in doc.xpath('//wps:ComplexData', namespaces= {'wps': 'http://www.opengis.net/wps/1.0.0'}))
+        _, self.tmppath = tempfile.mkstemp("." + self.getFileExtension("raster"))
+        f = open(self.tmppath, "w")
+        f.write(encodedText.decode('base64'))
+        f.close()
+        gdal.AllRegister()
+
+        exp_path = os.path.join(
+            self.getExpectedFileDir(), self.getExpectedFileName("raster")
+        )
+
+        try:
+            self.res_ds = gdal.Open(self.tmppath, gdal.GA_ReadOnly)
+        except RuntimeError as e:
+            self.fail("Response could not be opened with GDAL. Error was %s" % e)
+
+        try:
+            self.exp_ds = gdal.Open(expected_path, gdal.GA_ReadOnly)
+        except RuntimeError:
+            self.skipTest("Expected response in '%s' is not present" % exp_path)
+
+        # compare the size
+        self.assertEqual((self.res_ds.RasterXSize, self.res_ds.RasterYSize),
+                         (self.exp_ds.RasterXSize, self.exp_ds.RasterYSize))
+        # compare the band count
+        self.assertEqual(self.res_ds.RasterCount, self.exp_ds.RasterCount)
```

Comment:
The temporary files need to be deleted, otherwise they will clump up the tmp directory.